blocks/newsletter-signup/src/index.js

```diff
@@ -0,0 +1,33 @@
+import { registerBlockType } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+import './style.scss';
+import Edit from './edit';
+import metadata from './block.json';
+
+const smailyIcon = (
+	<svg
+		width="48"
+		height="48"
+		viewBox="0 0 48 48"
+		fill="none"
+		xmlns="http://www.w3.org/2000/svg"
+	>
+		<path
+			d="M39.9967 48H8.00318C3.58037 48 0 44.4019 0 39.9809V8C0 3.57895 3.58037 0 8.00318 0H39.9967C44.4196 0 47.9999 3.57895 47.9999 8V39.9809C48.0191 44.4019 44.4196 48 39.9967 48Z"
+			fill="#E91E63"
+		/>
+		<path
+			d="M26.9025 43.5791C26.2898 43.9427 25.6006 44.2106 24.8347 44.3829C24.0689 44.5551 23.3221 44.6508 22.5754 44.6508C21.0437 44.6508 19.5695 44.3063 18.1335 43.5982C17.2528 43.1771 16.4678 42.6413 15.7785 41.9905C14.8978 45.4355 11.7769 47.9618 8.0625 47.981H35.2312C35.8439 47.2346 36.38 46.3925 36.8203 45.4929C37.6819 43.7322 38.1223 41.6652 38.1223 39.3111C38.1223 36.8231 37.6436 34.8518 36.7055 33.359C35.7673 31.8853 34.5802 30.6987 33.1442 29.8183C31.7083 28.9379 30.1765 28.2489 28.53 27.7704C26.8834 27.292 25.3325 26.8135 23.9157 26.335C22.4797 25.8566 21.2926 25.2824 20.3545 24.5934C19.4163 23.9044 18.9376 22.9283 18.9376 21.646C18.9376 20.8422 19.1291 20.1724 19.5312 19.5982C19.9141 19.0432 20.431 18.5838 21.0629 18.2393C21.6947 17.8948 22.3648 17.646 23.0732 17.4929C30.1574 16.1915 31.8423 21.9523 31.8423 21.9523L38.084 13.0527C37.5479 12.5743 37.0118 12.1341 36.4374 11.7513L36.4183 11.7322C35.0972 10.8135 33.6612 10.1054 32.1295 9.6269C29.9276 8.93791 27.6492 8.61255 25.2942 8.61255C23.2456 8.61255 21.2735 8.89963 19.3397 9.47379C17.4251 10.048 15.721 10.9283 14.2468 12.0767C12.7725 13.2441 11.6046 14.6795 10.7047 16.402C9.82397 18.1245 9.3836 20.1341 9.3836 22.4116C9.3836 24.7657 9.86226 26.6795 10.8196 28.1149C11.7769 29.5503 12.9831 30.7178 14.4382 31.5982C15.8934 32.4786 17.4634 33.1867 19.1482 33.7035C20.8331 34.2202 22.4031 34.7369 23.8582 35.2728C25.3134 35.7896 26.5196 36.4212 27.4769 37.1676C28.4342 37.914 28.9129 38.9283 28.9129 40.1915C28.9129 40.957 28.7214 41.6078 28.3576 42.1819C28.013 42.7561 27.5152 43.2154 26.9025 43.5791Z"
+			fill="white"
+		/>
+	</svg>
+);
+
+registerBlockType( metadata.name, {
+	edit: Edit,
+	icon: smailyIcon,
+	title: __( 'Smaily Sign-Up Form', 'smaily' ),
+	description: __( 'Smaily newsletter subscription form.', 'smaily' ),
```

Comment:
Might need a name improvement here:
![image](https://github.com/user-attachments/assets/a0b345bb-1954-43aa-b348-8c2121f2c54e)

Reply:
Maybe *Opt-in subscribers directly to Smaily for seamless email marketing.*